### PR TITLE
Deferred unpack for memory and loading speed improvements in player and capture.

### PR DESCRIPTION
**Pull request by mkassner**
The PR shows WIP.

### CODE REVIEW

File: pupil_src/shared_modules/file_methods.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def close(self):
 def _load_object_legacy(file_path):
     file_path = os.path.expanduser(file_path)
     with open(file_path, 'rb') as fh:
-        data = pickle.load(fh, encoding='bytes')
+        data = pickle.load(fh, raw=True)
```
Comment (papr) [inline pupil_src/shared_modules/file_methods.py]: Is this change intentional?

Comment (mkassner) [reply pupil_src/shared_modules/file_methods.py]: new zmq api.

Comment (mkassner) [reply pupil_src/shared_modules/file_methods.py]: recommended change.

Comment (mkassner) [reply pupil_src/shared_modules/file_methods.py]: nvermind. you are right. This was a mistake.

File: pupil_src/shared_modules/file_methods.py [hunk 2]
```diff
@@ -93,3 +118,182 @@ def next_export_sub_dir(root_export_dir):
...
+        self._deser()
+        return item in self._data
+
+    def __iter__(self):
+        self._deser()
+        return iter(self._data)
+
+    def __unicode__(self):
+        self._deser()
+        return unicode(repr(self._data))
```
Comment (papr) [inline pupil_src/shared_modules/file_methods.py]: `unicode()` is not Python 3 function. I guess that the original code was meant to be written for Python 2.X. Can this function be removed?

File: pupil_src/launchables/player.py [hunk 3]
```diff
@@ -186,7 +186,7 @@ def on_drop(window, count, paths):
         def get_dt():
             return next(tick)
 
-        pupil_data_path = os.path.join(rec_dir, "pupil_data")
+        pupil_data_path = os.path.join(rec_dir, "pupil_data.pldata")
```
Comment (mkassner) [inline pupil_src/launchables/player.py]: please revert the extension change for now.

Comment (papr) [reply pupil_src/launchables/player.py]: It is just for testing purposes. The PR is still WIP.

File: pupil_src/shared_modules/player_methods.py [hunk 4]
```diff
@@ -58,7 +58,7 @@ def correlate_data(data, timestamps):
             # we might loose a data point at the end but we dont care
             break
 
-        if datum['timestamp'] <= ts:
+        if datum.get('timestamp', 0.) <= ts:
```
Comment (mkassner) [inline pupil_src/shared_modules/player_methods.py]: why make timestamp optional? I think we should require this. Data without timestamp can simply not be correlated...

**Comment by papr** [comment]
[`v1.8` Testing Procedure](https://hackmd.io/jTf2E5SARIWJmd_VXB9SmQ?view)
